edx-solutions/xblock-drag-and-drop-v2
# [MCKIN-2723] Use i18n runtime service to provide translations for HTML templates

Once the dependent PRs are merged, this change is all that's required to allow this XBlock to use its own translations for text found in HTML templates.

**JIRA tickets**: [MCKIN-7023](https://edx-wiki.atlassian.net/browse/MCKIN-7023), [OSPR-2289](https://openedx.atlassian.net/browse/OSPR-2289)

**Discussions**: [WL-230](https://openedx.atlassian.net/browse/WL-230), [YONK-879](https://openedx.atlassian.net/browse/YONK-879?focusedCommentId=307771&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-307771)

**Dependencies**:

* https://github.com/edx/xblock-utils/pull/48 - merged
* https://github.com/edx/xblock-utils/pull/49 - merged

**Screenshots**: 

**Before** 
these changes, the only text in DnDv2 HTML templates that would appear translated were the text that happened to be translated in edx-platform.
![before 1](https://user-images.githubusercontent.com/7556571/37516056-da821546-295c-11e8-8192-387bd59edf28.png)

**After** this change, DnDv2 template translations like [this one](https://github.com/edx-solutions/xblock-drag-and-drop-v2/blob/master/drag_and_drop_v2/templates/html/drag_and_drop_edit.html#L170-L174) appear in Studio and LMS:
![after 1](https://user-images.githubusercontent.com/7556571/37516104-f8e2c3dc-295c-11e8-89ed-f21fee13b860.png)

**Sandbox URL**: ([3f81e62](https://github.com/edx/edx-platform/pull/17670/commits/3f81e62f749b887cc50c3b9bd4d2b79c589d6a4a))

* LMS: https://pr17670.sandbox.opencraft.hosting/
* Studio: https://studio-pr17670.sandbox.opencraft.hosting/

**Merge deadline**: None

**Testing instructions**:

1. In Studio, create a course and add a Drag and Drop problem (e.g. [sandbox Drag and Drop](https://studio-pr17670.sandbox.opencraft.hosting/container/block-v1:OpenCraft+DnDv2+2018_1+type@vertical+block@a2b13b1603724df192e1c43dd3f1d70d?action=new#))
1. Visit the Studio URL with `/update_lang` appended, and submit `eo` as your selected Language Code.
1. Edit the Drag and Drop problem, and note that both [`blocktrans`](https://github.com/edx-solutions/xblock-drag-and-drop-v2/blob/master/drag_and_drop_v2/templates/html/drag_and_drop_edit.html#L170-L174) and [`trans`](https://github.com/edx-solutions/xblock-drag-and-drop-v2/blob/master/drag_and_drop_v2/templates/html/drag_and_drop_edit.html#L165) template tag strings appear translated.

**Author notes and concerns**:

1. Translations in the Javascript templates are currently not working at all, and will be addressed by a future change.  This issue can be seen in the LMS (student view), and the Studio view, when adding/removing zones.
1. The [process for extracting translations for this XBlock is still quite manual](https://github.com/edx-solutions/xblock-drag-and-drop-v2#i18n-compatibility), and so this change does not attempt to correct the approach, or to provide any missing translations.

**Reviewers**
- [ ] edx-solutions reviewer[s] TBD
